testproject/Assets/Tests/Runtime/MultiprocessRuntime/TestCoordinator.cs

```diff
@@ -37,9 +40,25 @@ public class TestCoordinator : NetworkBehaviour
     public static List<ulong> AllClientIdsWithResults => Instance.m_TestResultsLocal.Keys.ToList();
     public static List<ulong> AllClientIdsExceptMine => NetworkManager.Singleton.ConnectedClients.Keys.ToList().FindAll(client => client != NetworkManager.Singleton.LocalClientId);
 
-    private void Awake()
+    // Multimachine support
+    private static int s_ProcessId;
+    public static string Rawgithash;
+    public static ConfigurationType ConfigurationType;
+    private string m_ConnectAddress = "127.0.0.1";
+    public static string Port = "3076";
+    private bool m_IsClient;
+
+    public void Awake()
     {
-        MultiprocessLogger.Log("Awake");
+        s_ProcessId = Process.GetCurrentProcess().Id;
+        MultiprocessLogger.Log($"Awake {s_ProcessId}");
+        ReadGitHashFile();
+
+        // There are two possible ways for configuration data to be set
+        // 1. From a webapi
+        // 2. From a text file resource
+        ConfigureViaWebApi();
```

Comment:
now I'm confused, this works without waiting for the configuration? I would guess this runs async, but then your Start would run right after without waiting for the config, using the default connect address, no?

Reply:
Yes,  you are right, this is a problem. I've made an update to move the initialization of the Client to the Update method after the properties have been set.

Reply:
You'd have the same issue in an update as in a start no? You're just waiting a bit longer, but if your config server waits even longer, that'll still cause issues?
Why not just wait in the awake?

Reply:
ok I think I understand a bit more the flow and why it works at the moment.
--> So because this is happening client side, the client won't start until it receives those parameters, and so won't start receiving test calls until it's connected.
Putting this in Update just calls for having bugs in the future though ("why is my StartClient called multiple times"?) and isn't super readable.
When I mentioned in my previous comment about reducing our "Wait" usage as much as possible, I meant in the full callstack, not sprinkling Wait everywhere. The initial callsite could still use Wait though, no? (and read the wrapped exception if there's one?)

Reply:
Now I'm confused.
Where do you see Wait sprinkled everywhere? And why would StartClient get called more than once?